pkg/asset/cluster/cluster.go

```diff
@@ -81,22 +83,24 @@ func (c *Cluster) Generate(parents map[asset.Asset]*asset.State) (*asset.State,
 
 	stateFile, err := terraform.Apply(tmpDir)
 	if err != nil {
-		// we should try to fetch the terraform state file.
-		log.Errorf("terraform failed: %v", err)
+		err = fmt.Errorf("terraform failed: %v", err)
```

Comment:
If `stateFile` is empty, I think we want to [bail here][1] instead of continuing on to attempt to read it.

[1]: https://github.com/openshift/installer/pull/375#discussion_r221441269

Reply:
stateFile is never empty; it is predetermined file location.https://github.com/openshift/installer/blob/be8144a85d137bb99278fa9f54e6bfcdbf78cf36/pkg/terraform/terraform.go#L24

Reply:
> `stateFile` is never empty; it is predetermined file location.

Are we comfortable making that guarantee part of `Apply`'s public API?  If so, can we update the `Apply` godocs to make that explicit?  If not, can we check here anyway (protecting against future `Apply` which might return empty strings)?

Reply:
I updated the docs to reflect that the filename will always be returned.